internal/store/firestore.go

```diff
@@ -0,0 +1,154 @@
+package firestore
+
+import (
+	"context"
+	"fmt"
+
+	"cloud.google.com/go/firestore"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"pantheon.io/edrt-policy-docs-functions/internal/types"
+)
+
+const hostnameCollectionName = "hostnames"
+const hostnameMetadataCollectionName = "hostnameMetadata"
+const edgeLogicCollectionName = "edgelogic"
+
+type Firestore struct {
+	logger    *logrus.Logger
+	projectId string
+}
+
+func New(logger *logrus.Logger, projectId string) Firestore {
+	return Firestore{
+		logger:    logger,
+		projectId: projectId,
+	}
+}
+
+func (f *Firestore) Read(ctx context.Context) interface{} {
+	client, err := firestore.NewClient(ctx, f.projectId)
+	if err != nil {
+		f.logger.Fatalf("Unable to create firestore client %s", err)
+	}
+
+	collection := client.Collection("foo")
+	doc := collection.Doc("bar")
+	data, err := doc.Get(ctx)
+	if err != nil {
+		f.logger.Fatalf("Unable to get document %s", err)
```

Comment:
It is better to return errors or handler them gracefully instead of terminating the application immediately. Under the hood, `logger.Fatalf` uses syscall and exits:
```go
func (logger *Logger) Fatalf(format string, args ...interface{}) {
	logger.Logf(FatalLevel, format, args...)
	logger.Exit(1)
}
```

From the cloud function perspective, it means the function can be retried. I guess we do not want to retry again and again if the document not found. We can handle this case properly.

Reply:
Yes, good point. This function was actually my first shot at reading from firestore and it's just some dummy data. This function is not meant to go live. The other functions in this file that actually read a fake pdoc from my test firestore instance are returning errors instead of killing the process.

I do still use logger.Fatalf in the `cmd/cli` file, but I think that's acceptable since it's not running in a cloud function.

Reply:
This function has been removed